web/components/application/hacker/Hacker.js

```diff
@@ -32,19 +33,34 @@ const HackerApplication = (props) => {
     onPageNext,
     onHackerApplicationChange,
     cancelHackerApplication,
+    isNextButtonEnabled,
+    enableNextButton,
+    disableNextButton,
   } = props;
 
   const indexToPage = {
     0: (
       <PageOne
         hackerApplication={hackerApplication}
         onHackerApplicationChange={onHackerApplicationChange}
+        enableNextButton={enableNextButton}
```

Comment:
this seems brittle... why not a single callback like `fieldsAreValid()` to indicate that the page is next-able?

Reply:
This is not to "indicate" whether or not the page is "next-able". This is an action/function to actually enable the next button.

Reply:
i might be missing something, but isnt enabling the next button just a roundabout way of indicating that this page is `next-able` (if "next-able", enable my button - if not "next-able", disable my button)?

Reply:
Oh are you asking why we need both `enableNextButton` and `disableNextButton` instead of just `enableNextButton`/`fieldsAreValid`?

If that's what you're asking, we also need to be able to disable the next button when the form in the page is no longer valid.

Reply:
but if you have:

```js
fieldsAreValid(valid? bool)
```

child:

```js
// insert destructuring lol
const valid = this.state.isValid; // true or false
fieldsAreValid(valid)
```

parent:

```js
fieldsAreValid(valid => if (valid) { enableButton(); } else { disableButton(); }
// this is definitely not valid syntax
```

Reply:
oh yeah that's doable. but why do you exactly prefer it this way?

Reply:
less things to pass around, and imo having two functions for what is essentially passing back a boolean state might be a bit much